atomic_reactor/util.py

```diff
@@ -619,10 +619,21 @@ def __init__(self, secret_path):
         :param secret_path: str, dirname of .dockercfg location
         """
 
-        self.json_secret_path = os.path.join(secret_path, '.dockercfg')
+        if os.path.exists(os.path.join(secret_path, '.dockercfg')):
+            self.json_secret_path = os.path.join(secret_path, '.dockercfg')
+        elif os.path.exists(os.path.join(secret_path, '.dockerconfigjson')):
+            self.json_secret_path = os.path.join(secret_path, '.dockerconfigjson')
+        else:
+            raise RuntimeError("The registry secret was not found on the filesystem, "
+                               "either .dockercfg or .dockerconfigjson are supported")
         try:
             with open(self.json_secret_path) as fp:
                 self.json_secret = json.load(fp)
+
+            # If the auths key exist then we have a dockerconfigjson secret.
+            if self.json_secret.get('auths') is not None:
+                self.json_secret = self.json_secret.get('auths')
```

Comment:
The comment doesn't technically describe what the code does ('if auths key exists' vs `if [value] is not None`). Maybe re-write as:
```
if 'auths' in self.json_secret:
    self.json_secret = self.json_secret['auths']
```